validator/remote/src/main/java/tech/pegasys/teku/validator/remote/BeaconNodeReadinessManager.java

```diff
@@ -192,9 +198,10 @@ private void processReadyResult(final boolean isPrimaryNode) {
     if (!isPrimaryNode) {
       return;
     }
+    // Filtering of duplicates if needed happens on receiver's side
+    beaconNodeReadinessChannel.onPrimaryNodeBackReady();
```

Comment:
I don't understand the purpose of this change. At this point the primary node is not yet confirmed **back** ready (could be already marked as ready) so why should we notify the subscribers before checking that its readiness status changed?

Reply:
The point is that we could have inconsistency between Manager and subscribers. We send readiness only once and if subscriber failed to process it correctly, it will never receive another readiness keeping running on failover

Reply:
Ok I see your point. Wouldn't it then be better to rename `onPrimaryNodeBackReady` to simply `onPrimaryNodeReady` since there is no _back ready_ notion anymore?

Reply:
Nice suggestion! Updated